### PR TITLE
SDKS-3408 Self device management implementation

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket

[SDKS-3408](https://pingidentity.atlassian.net/browse/SDKS-3408)

# Description

Self Device Management Service
```kotlin
val oathDevices: List<OathDevice> = userRepository.oathDevice()
val pushDevices: List<PushDevice> = userRepository.pushDevice()
val webAuthnDevices: List<WebAuthnDevice> = userRepository.webAuthnDevice()
val bindingDevice: List<BindingDevice> = userRepository.bindingDevice()
val profileDevice: List<ProfileDevice> = userRepository.profileDevice()

//Update Device Name
webAuthnDevices[0].deviceName = "New Device Name"
userRepository.update(webAuthnDevices[0])

//Delete Device
userRepository.delete(webAuthnDevices[0])
```

Sample App Update:
CleanShot 2024-10-09 at 11 58 27@2x

CleanShot 2024-10-09 at 11 59 30@2x

### CODE REVIEW

File: forgerock-auth/src/main/java/org/forgerock/android/auth/Config.java [hunk 1]
```diff
@@ -169,7 +169,7 @@ public static Config getInstance() {
         return mInstance;
     }
 
-    ServerConfig getServerConfig() {
+    public ServerConfig getServerConfig() {
```
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/Config.java:172]: Add doc

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Just a minor comment to be addressed

**Review by spetrov — DISMISSED** [review]
Looks good to me! 👍🏻

File: forgerock-auth/src/main/java/org/forgerock/android/auth/Config.java (hunk 1 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/Config.java:176]: Updated.

**Review by spetrov — APPROVED** [review]

**Pull request merged**